RELEASE_NOTES.md

```diff
@@ -145,4 +145,4 @@
 
 ## Bug Fixes
 
-<!-- Here goes notable bug fixes that are worth a special mention or explanation -->
+* `Timer`: Fix bug where if `reset()` was called while the timer was being waited on, the timer would not be reset and still wait for the original timeout.
```

Comment:
I'd suggest either `where` or `if` but not both
```suggestion
* `Timer`: Fix bug if `reset()` was called while the timer was being waited on, the timer would not be reset and still wait for the original timeout.
```

Reply:
That was actually written by copilot, I thought it could do English better than me :P

ChatGPT says:

![image](https://github.com/frequenz-floss/frequenz-channels-python/assets/1031485/ff8c36d4-06fa-4b8e-9540-bf381309c576)

All robots seem to be on my side :laughing: but let me know if you still believe the `where` is incorrect. Of course I trust you more than the robots...

Reply:
I'm afraid the robot is right :disappointed: 
My suggestion sounded like fixing the bug is conditional

Reply:
"fix bug where if" sounds more accurate, but it might be clearer to reorder the clauses, to keep the two words apart: "where x was failing if done while y was going on"

Fix bug that was causing calls to `reset` to not reset the timer, if the timer was already being awaited.

Reply:
OK, the robot still agrees with me (which actually means agreeing with the other robot).

![image](https://github.com/frequenz-floss/frequenz-channels-python/assets/1031485/c7aa8269-a189-4d74-8880-67f227e9629e)

But I amended the commit to use @shsms proposed phrasing anyway, as it seems it was not clear enough for humans :smile: